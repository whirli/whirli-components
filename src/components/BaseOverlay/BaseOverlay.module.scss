@import '../../styles/Breakpoints.module.scss';
@import '../../styles/Spacers.scss';

// default styles
.overlay {
    padding: $spacer2Xl 0;
    box-sizing: border-box;
    z-index: 1000;
    height: 100%;
    width: 100%;
    position: fixed;
    inset: 0;
    display: flex;
    justify-content: center;
    align-items: center;
}

.state-closed {
    visibility: hidden;
    opacity: 0;
    transition: ease-in-out opacity 0.5s 0.2s, ease-in-out visibility 1s;
}

.state-open {
    visibility: visible;
    opacity: 1;
    transition: ease-in-out opacity 0.5s, ease-in-out visibility 0s;
}

.overlay__content {
    display: flex;
    justify-content: center;
    align-items: center;
}

.state-closed .overlay__content {
    visibility: hidden;
    opacity: 0;
    transition: ease-in-out opacity 0.5s, ease-in-out visibility 0.5s;
}

.state-open .overlay__content {
    visibility: visible;
    opacity: 1;
    transition: ease-in-out opacity 0.5s, ease-in-out visibility 0s;
}

.color-default {
    background-color: rgba(0, 0, 0, 0.2);
}
// breakpoint styles
$packageBreakpointStyles: ();