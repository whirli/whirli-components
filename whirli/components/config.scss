@import '@whirli-components/styles/Helpers.scss';

$breakpointPrefix: BREAKPOINT_;

$breakpoints: (
    tablet: (
        prefix: t,
        min-width: 672px,
    ),
    small_desk: (
        prefix: sd,
        min-width: 1056px
    ),
    large_desk: (
        prefix: ld,
        min-width: 1312px
    ),
    huge_desk: (
        prefix: hd,
        min-width: 1584px
    ),
);

:export {
    prefix: $breakpointPrefix;
    #{$breakpointPrefix}default: 0px;
    @each $name, $value in $breakpoints {
        #{$breakpointPrefix}#{$name}: map-deep-get($breakpoints, #{$name}, 'min-width')
    }
}