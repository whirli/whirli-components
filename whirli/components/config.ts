import BaseAside from './BaseAside/BaseAside.config';
import BaseButton from './BaseButton/BaseButton.config';
import BaseColumn from './BaseColumn/BaseColumn.config';
import BaseContainer from './BaseContainer/BaseContainer.config';
import BaseGrid from './BaseGrid/BaseGrid.config';
import BaseImage from './BaseImage/BaseImage.config';
import BaseMain from './BaseMain/BaseMain.config';
import BaseSection from './BaseSection/BaseSection.config';
import BaseSpacer from './BaseSpacer/BaseSpacer.config';
import BaseText from './BaseText/BaseText.config';

export default {
  BaseAside,
  BaseButton,
  BaseColumn,
  BaseContainer,
  BaseGrid,
  BaseImage,
  BaseMain,
  BaseSection,
  BaseSpacer,
  BaseText,
};
