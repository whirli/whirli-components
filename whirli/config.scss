@import '@whirli-components/src/styles/Helpers.scss';

$breakpoints: (
    tablet: (
        prefix: t,
        min-width: 48rem,
    ),
    desktop: (
        prefix: d,
        min-width: 80rem,
    )
);

:export {
    @each $name, $value in $breakpoints {
        BREAKPOINT_#{to-upper-case(#{$name})}: map-deep-get($breakpoints, #{$name}, 'min-width')
    }
}