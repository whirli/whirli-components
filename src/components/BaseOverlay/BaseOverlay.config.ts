// @ts-ignore
import { PropType } from 'vue';
import * as PropHelpers from '../../helpers/props';
import { ComponentStyles, ComponentProps } from '../../@types/components';

// Local config
import config from '@whirli-local/components/BaseOverlay/BaseOverlay.config';

// Package config
import { PropKeys, PropValues } from './BaseOverlay.constants';
import type { PropColor, PropState } from './BaseOverlay.types';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
  [PropKeys.COLOR]: {
    generateBreakpoints: true,
    classes: {
      default: 'color-default',
      ...config.styles[PropKeys.COLOR]?.classes,
    } as Record<PropColor, string>,
  },
  [PropKeys.STATE]: {
    classes: {
      closed: 'state-closed',
      open: 'state-open',
      ...config.styles[PropKeys.STATE]?.classes,
    } as Record<PropState, string>,
  },
};

export const ConfigProps: ComponentProps = {
  [PropKeys.COLOR]: {
    required: false,
    type: String as PropType<PropColor>,
    default: 'default',
    ...PropHelpers.breakpointStylePropValidation(ConfigStyles, PropKeys.COLOR),
  },
  [PropKeys.STATE]: {
    required: false,
    type: String as PropType<PropState>,
    default: 'closed',
    ...PropHelpers.stylePropValidation(ConfigStyles, PropKeys.STATE),
  },
  ...config.props,
};
