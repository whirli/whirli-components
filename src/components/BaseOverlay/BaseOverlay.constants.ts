// Local config
import config from '@whirli-local/components/BaseOverlay/BaseOverlay.config';

// Helpers
import { getLocalPropValues } from '@whirli-components/helpers/props';

export const PropKeys: Record<string, string> = {
  COLOR: 'color',
  STATE: 'state',
};

const localValues = getLocalPropValues(PropKeys, config);

export const PropValues: Record<string, string[]> = {
  [PropKeys.COLOR]: ['default', ...localValues[PropKeys.COLOR]],
  [PropKeys.STATE]: ['open', 'closed'],
};
