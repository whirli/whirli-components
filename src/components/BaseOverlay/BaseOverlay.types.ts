import { PropKeys, PropValues } from './BaseOverlay.constants';

const propColor: string[] = PropValues[PropKeys.COLOR];
const propState: string[] = PropValues[PropKeys.STATE];

export type PropColor = typeof propColor[number];
export type PropState = typeof propState[number];
