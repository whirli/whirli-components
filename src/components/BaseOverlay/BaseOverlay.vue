<template>
  <div :class="wrapperClasses">
    <div :class="contentClasses">
      <slot />
    </div>
  </div>
</template>

<script setup lang="ts">
// Vue
import { computed } from '@composition';
import { ComputedRef } from '@composition';

// Styles
import styles from '@whirli-local/components/BaseOverlay/BaseOverlay.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseOverlay.config';

const ComponentStyles = ConfigStyles;

const props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const wrapperClasses: ComputedRef<string[]> = computed(() => {
  return [styles.overlay, ...makeClasses(ComponentStyles, props, styles)];
});

const contentClasses = [styles['overlay__content']];
</script>
