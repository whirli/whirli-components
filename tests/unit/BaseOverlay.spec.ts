import { shallowMount } from '@vue/test-utils';

import { propReturnsClass } from '../helpers';

import BaseOverlay from '@/components/BaseOverlay/BaseOverlay.vue';
import { PropKeys } from '@/components/BaseOverlay/BaseOverlay.constants';
import { ConfigStyles } from '@/components/BaseOverlay/BaseOverlay.config';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let wrapper: any;
const text = 'Hello World';

describe('BaseOverlay.vue - Single classes', () => {
  beforeEach(() => {
    wrapper = shallowMount(BaseOverlay, {
      slots: {
        default: text,
      },
    });
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the correct text', () => {
    expect(wrapper.text()).toMatch(text);
  });

  it('is the correct color', () => {
    expect(propReturnsClass(wrapper, ConfigStyles, PropKeys.COLOR, 'color-default')).toBe(true);
  });
});
