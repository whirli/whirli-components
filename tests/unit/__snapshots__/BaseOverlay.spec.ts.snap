// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseOverlay.vue - Single classes renders the component 1`] = `
<div
  class=""
>
  <div
    class=""
  >
    
    Hello World
    
  </div>
</div>
`;
